kernel/environ.mli

```diff
@@ -283,6 +283,32 @@ val template_polymorphic_ind : inductive -> env -> bool
 val template_polymorphic_variables : inductive -> env -> Univ.Level.t list
 val template_polymorphic_pind : pinductive -> env -> bool
 
+(** {6 Name quotients} *)
+
+module type QNameS =
+sig
+  type t
+  val equal : env -> t -> t -> bool
```

Comment:
Having the full `env` here may be too heavy of a choice future-wise; IMHO I'd be better to make it a different type for now as to have more flexibility.

Reply:
Actually I thought about doing that at first, and decided against judging from the fact that environments are pervasive in the upper layers. The kernel shouldn't use these functions anyways, and we already pass an environment everywhere outside, so it would add more clutter than anything. Furthermore, if ever we fix the module system to get applicative functors, alias expressions would be arbitrary λ-terms, requiring a full-blown environment to decide equality.